tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,415 @@
+            if not events:
+                continue
+            events = AzureEvents.get_event_operations(events)
+            if operation_name in events:
+                self.scheduler.add_job(self.run_policy,
+                                       id=k + event['id'],
+                                       name=k,
+                                       args=[v['policy'],
+                                             event,
+                                             None],
+                                       misfire_grace_time=60 * 3)
+
+    def run_policy(self, policy, event, context):
+        try:
+            policy.push(event, context)
+        except Exception as e:
+            log.error(
+                "Exception running policy: %s error: %s",
+                policy.name, e)
+
+    def prepare_queue_storage(self, queue_resource_id, queue_name):
+        """
+        Create a storage client using unusual ID/group reference
+        as this is what we require for event subscriptions
+        """
+        storage_client = self.session.client('azure.mgmt.storage.StorageManagementClient')
+
+        account = storage_client.storage_accounts.get_properties(
+            ResourceIdParser.get_resource_group(queue_resource_id),
+            ResourceIdParser.get_resource_name(queue_resource_id))
+
+        Storage.create_queue_from_storage_account(account,
+                                                  queue_name,
+                                                  self.session)
+        return account
+
+    @staticmethod
+    def has_required_params():
+        required = [
+            constants.ENV_CONTAINER_POLICY_STORAGE,
+            constants.ENV_CONTAINER_EVENT_QUEUE_NAME,
+            constants.ENV_CONTAINER_EVENT_QUEUE_ID
+        ]
+
+        missing = [r for r in required if os.getenv(r) is None]
+
+        if missing:
+            log.error('Missing REQUIRED environment variable(s): %s' % ', '.join(missing))
+            return False
+
+        return True
+
+    @staticmethod
+    def build_options():
+        """
+        Accept some CLI/Execution options as environment
+        variables to apply global config across all policy
+        executions.
+        """
+        output_dir = os.environ.get(constants.ENV_CONTAINER_OPTION_OUTPUT_DIR)
+
+        if not output_dir:
+            output_dir = tempfile.mkdtemp()
+            log.warning('Output directory not specified.  Using directory: %s' % output_dir)
+
+        config = Config.empty(
+            **{
+                'log_group': os.environ.get(constants.ENV_CONTAINER_OPTION_LOG_GROUP),
+                'metrics': os.environ.get(constants.ENV_CONTAINER_OPTION_METRICS),
+                'output_dir': output_dir
+            }
+        )
+
+        return Azure().initialize(config)
+
+    @staticmethod
+    def has_yaml_ext(filename):
+        return filename.lower().endswith(('.yml', '.yaml'))
+
+
+if __name__ == "__main__":
+    Host()
```

Comment:
using click as a cli library would allow for picking up config from cli commands and env vars, and documenting them directly on the cli also provides for discoverability.

Reply:
Yeah this is an excellent idea, along with the recommendation to decouple the config/cli from the host class.